addons/udes_security/controllers/main.py

```diff
@@ -87,17 +96,36 @@ def _get_menu_info_from_action_id(self, action_id):
         """
         request.env.cr.execute(query, (action_id,))
         result = request.env.cr.fetchone()
-        if not result:
-            return [], []
-        (parent_path, relation_table, inverse, action_id) = result
-        # Get ids in order from parent --> child of the menus
-        parent_path_ids = [int(str_menu_id) for str_menu_id in parent_path.split("/")[:-1]]
-        action_group_ids = []
-        if relation_table and inverse:
-            action_group_ids = self._get_object_group_ids(action_id, relation_table, inverse)
+        if result:
+            (parent_path, relation_table, inverse, action_id) = result
+            # Get ids in order from parent --> child of the menus
+            parent_path_ids = [int(str_menu_id) for str_menu_id in parent_path.split("/")[:-1]]
+            if relation_table and inverse:  # Some ir.actions* models don't have groups_id column
+                action_group_ids = self._get_object_group_ids(action_id, relation_table, inverse)
+        else:
+            # Certain actions don't have a menu, but we still need to determine the model
+            # which the action_id is for to get its groups. This approach is slower.
+            # This will happen rarely, so not concerned about optimising.
+            models = [
+                "ir.actions.act_window",
+                "ir.actions.server",
+                "ir.actions.client",
+                "ir.actions.report",
+                "ir.actions.act_url",
+            ]
+            for model in models:
+                action = request.env[model].search([("id", "=", action_id)])
```

Comment:
Suggestion: Add `order=id` since ordering doesn't matter.

Reply:
Although follow on question... is it possible to have two `ir.actions.act_window` records with the same action but different groups?

Reply:
Users --> Change password wizard
without prefetch change
```action = request.env[model].search([("id", "=", action_id)])```
Hook took 0.010319232940673828
with prefetch change
```action = request.env[model].with_context(prefetch_fields=False).search([("id", "=", action_id)])```
Hook took 0.01175546646118164
with order=id change too
```action = request.env[model].with_context(prefetch_fields=False).search([("id", "=", action_id)], order="id")```
Hook took 0.010631799697875977

Reply:
Yep let's not worry about prefetching then.

Just suggest to add order by ID as this will be simpler than the ordering set on the models.

Reply:
Seems like it takes longer to override the default search? Anyway these are very small timings, all good will resolve.

Reply:
Added order="id" and also a check for groups_id field existing on the line below (in 11 and 14) which i forgot.